libopendavinci/src/odcore/reflection/MessageResolver.cpp

```diff
@@ -0,0 +1,211 @@
+/**
+ * OpenDaVINCI - Portable middleware for distributed components.
+ * Copyright (C) 2017 Christian Berger
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include <iostream>
+
+#ifdef HAVE_DL
```

Comment:
I know but the dlfcn is not present on all platforms (yet) and I do not want to pull in Boost on that level.